bin/http-signature-header-js

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env node
+
+const program = require('commander');
+const httpSigs = require('../lib/');
+const fs = require('fs');
+const util = require('util');
+const httpMessageParser = require('http-message-parser');
+
+// this can be use to await paths to private keys and other files.
+// also can get be used to dereference keyId in middleware
+const readFile = util.promisify(fs.readFile);
+
+/**
+ * Simple wrapper around node's process.stdin.
+ * This allows us to get the result of cat and other commands.
+ *
+ * @param {string} [encoding='utf8'] - The http message's encoding.
+ *
+ * @returns {Promise<string>} Should return the http message.
+ */
+async function getStdIn(encoding = 'utf8') {
+  const {stdin} = process;
+  let message = '';
+  return new Promise((resolve, reject) => {
+    try {
+      stdin.setEncoding(encoding);
+      stdin.on('readable', () => {
+        let chunk;
+        while((chunk = stdin.read())) {
+          message += chunk;
+        }
+      });
+      stdin.on('end', () => resolve(message));
+    }
+    catch(e) {
+      reject(e);
+    }
+  });
+}
+
+async function getHttpMessage() {
+  const httpMessage = await getStdIn();
+  if(!httpMessage) {
+    throw new Error('An Http Message must be passed to stdin.');
+  }
+  // this will create a request or response object
+  // similar to node's default request object.
+  const request = httpMessageParser(httpMessage);
+  return request;
+}
+
+function makeHttpMessageOutgoing({status = '200 OK', version = '1.1', headers = {}}) {
+  let message = `HTTP/${version} ${status}`;
+  for(const key in headers) {
+    let value = headers[key];
+    if(Array.isArray(value)) {
+      value = value.join(',');
+    }
+    message += `${key}: ${value}`;
+  }
+  return message;
+}
+
+program
+  .command('c14n')
+  .alias('C')
+  .action(async () => {
+    try {
+      const {headers} = program;
+      const requestOptions = await getHttpMessage();
+      if(!headers) {throw new Error('--headers required');}
+      const includeHeaders = program.headers ? program.headers.split(',') : '';
+      const result = httpSigs.
+        createSignatureString({includeHeaders, requestOptions});
+      console.log(JSON.stringify(result));
+      process.exit(0);
+    } catch(e) {
+      throw e;
+      process.exit(1);
+    }
+  });
+
+program
+  .command('sign')
+  .alias('S')
+  .option('-s, --signature <signature>', 'Signature.')
+  .action(async cmd => {
+    try {
+      const {signature} = cmd;
+      const {
+        headers,
+        keyId,
+        privateKey,
+        privateKeyType,
+        algorithm,
+        created,
+        expires
+      } = program;
+      const privateKeyFile = await readFile(privateKey);
+      const includeHeaders = headers ? program.headers.split(',') : '';
```

Comment:
Why are you splitting on comma. You should be splitting on a space character instead. The command line option is going to be something like this:

`--headers "(request-target) (created) host digest content-length"`